bundles/org.openhab.binding.tado/src/main/java/org/openhab/binding/tado/internal/handler/TadoBatteryChecker.java

```diff
@@ -69,25 +79,8 @@ private synchronized void refreshZoneList() {
 
     public State getBatteryLowAlarm(long zoneId) {
         refreshZoneList();
-        boolean hasBatteryStateValue = false;
-        if (zoneList != null) {
-            // logger.debug("Fetching battery state for ZoneId {}", zoneId);
-            for (Zone zone : zoneList) {
-                if (zoneId == zone.getId()) {
-                    for (ControlDevice device : zone.getDevices()) {
-                        String batteryState = device.getBatteryState();
-                        if (batteryState != null) {
-                            if (!batteryState.equals("NORMAL")) {    
-                                return OnOffType.ON;
-                            }
-                            hasBatteryStateValue = true;
-                        }
-                    }
-                    break;
-                }
-            }
-        }
-        return hasBatteryStateValue ? OnOffType.OFF : UnDefType.UNDEF;
+        State state = zoneList.get(zoneId);
```

Comment:
```suggestion
        return zoneList.getorDefault(zoneId, OnOffType.OFF);
```

Reply:
or probably default to `UndefType.UNDEF` as we don't know if `ON` or `OFF` is correct.

Reply:
Agreed to your two points. I was not aware of the getOrDefault() method, so I definitely learned something useful, thank you.